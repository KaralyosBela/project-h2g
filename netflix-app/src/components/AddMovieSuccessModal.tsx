import classes from "./AddMovieSuccessModal.module.css";
import {AiFillCheckCircle} from "react-icons/ai"
import {CgClose} from "react-icons/cg"

interface Props {
  close: () => void
}

export const AddMovieSuccessModal: React.FC<Props> = ({ close }) => {
  return (
    <>
      <div className={classes.overlay} onClick={close} />
      <div className={classes.successModal}>
        <div className={classes.icon}><AiFillCheckCircle size={80}/></div>
        <h1 className={classes.title}>CONGRATULATIONS!</h1>
        <div className={classes.subtitle}>
          The movie has been added to the <br /> database successfully
        </div>
        <div className={classes.close} onClick={close}>              
        <CgClose size={30}/>
        </div>
      </div>
    </>
  );
};
