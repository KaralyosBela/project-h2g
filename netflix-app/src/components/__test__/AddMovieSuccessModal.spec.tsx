import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { AddMovieSuccessModal } from "../AddMovieSuccessModal";

const mockHide = jest.fn();

describe("Add Movie Succedd Modal", () => {

  it("should render the component", () => {
    const { container } = render(<AddMovieSuccessModal close={mockHide} />);
    expect(container).toMatchSnapshot();
  });

  it("should call the hide function from the prop", () => {
    render(<AddMovieSuccessModal close={mockHide} />);

    const overlay = screen.getByTestId("overlay");
    fireEvent.click(overlay);

    expect(mockHide).toBeCalledTimes(1);
  });

});
