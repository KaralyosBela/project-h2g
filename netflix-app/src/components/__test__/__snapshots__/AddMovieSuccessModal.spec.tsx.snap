// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Movie Succedd Modal should render the component 1`] = `
<div>
  <div
    class="overlay"
    data-testid="overlay"
  />
  <div
    class="successModal"
  >
    <div
      class="icon"
    >
      <svg
        fill="currentColor"
        height="80"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </div>
    <h1
      class="title"
    >
      CONGRATULATIONS!
    </h1>
    <div
      class="subtitle"
    >
      The movie has been added to the 
      <br />
       database successfully
    </div>
    <div
      class="close"
    >
      <svg
        fill="none"
        height="30"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.2253 4.81108C5.83477 4.42056 5.20161 4.42056 4.81108 4.81108C4.42056 5.20161 4.42056 5.83477 4.81108 6.2253L10.5858 12L4.81114 17.7747C4.42062 18.1652 4.42062 18.7984 4.81114 19.1889C5.20167 19.5794 5.83483 19.5794 6.22535 19.1889L12 13.4142L17.7747 19.1889C18.1652 19.5794 18.7984 19.5794 19.1889 19.1889C19.5794 18.7984 19.5794 18.1652 19.1889 17.7747L13.4142 12L19.189 6.2253C19.5795 5.83477 19.5795 5.20161 19.189 4.81108C18.7985 4.42056 18.1653 4.42056 17.7748 4.81108L12 10.5858L6.2253 4.81108Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
